src/TheNextWeek/main.cc

```diff
@@ -235,163 +304,44 @@ hittable_list final_scene() {
         boxes2.add(make_shared<sphere>(point3::random(0,165), 10, white));
     }
 
-    objects.add(make_shared<translate>(
+    world.add(make_shared<translate>(
         make_shared<rotate_y>(
             make_shared<bvh_node>(boxes2, 0.0, 1.0), 15),
             vec3(-100,270,395)
         )
     );
-
-    return objects;
 }
 
 
-color ray_color(const ray& r, const color& background, const hittable& world, int depth) {
-    hit_record rec;
-
-    // If we've exceeded the ray bounce limit, no more light is gathered.
-    if (depth <= 0)
-        return color(0,0,0);
-
-    // If the ray hits nothing, return the background color.
-    if (!world.hit(r, interval(0.001, infinity), rec))
-        return background;
-
-    ray scattered;
-    color attenuation;
-    color emitted = rec.mat_ptr->emitted(rec.u, rec.v, rec.p);
-
-    if (!rec.mat_ptr->scatter(r, rec, attenuation, scattered))
-        return emitted;
-
-    return emitted + attenuation * ray_color(scattered, background, world, depth-1);
+void default_scene(scene& scene_desc) {
+    final_scene(scene_desc);
+    scene_desc.image_width       = 400;
+    scene_desc.samples_per_pixel = 250;
+    scene_desc.max_depth         = 4;
 }
 
 
-int main() {
-
-    // Image
-
-    auto aspect_ratio = 16.0 / 9.0;
-    int image_width = 400;
-    int samples_per_pixel = 100;
-    int max_depth = 50;
-
-    // World
+void main() {
+    scene scene_desc;
 
-    hittable_list world;
+    scene_desc.background = color(0.70, 0.80, 1.00);
```

Comment:
This probably belongs in the scene function?

Most scenes redefine background to black, so move this line of code into those that don't.

Reply:
oh, wait. I see more `scene_desc` changes below. I guess it's fine for them to be outside of scene function calls.

I do think that these `scene_desc` changes should probably be placed _below_ the `switch` statement

Reply:
These statements effectively set up default scene values for this program. Subsequent scene modifying functions may either use these defaults or override them.